### PR TITLE
FHIR-38850, FHIR-38788, FHIR-39307 Negation profile changes. ICF JIRA CQF-1145

### PR DESCRIPTION
**Pull request by dand9959**
Commit 56cace7c18d5b11f2dbbf915181545e1b1b9ce64

**https://jira.hl7.org/browse/FHIR-38850**  

MedicationDispenseNotDone renamed MedicationDispenseDeclined  
ObservationNotDone renamed ObservationCancelled  
TaskNotDone rename TaskRejected  

**https://jira.hl7.org/browse/FHIR-38788**   

Communication - removed "not done" from code values  
Immunization - removed "not done" from code values  
MedicationAdministration - removed "not done" from code values  
MedicationDispense - removed "declined" from code values  
Observation - removed "cancelled" from code values  
Procedure - removed "not done" from code values  
Task - remvoed "Rejected" from code values  
Service - removed "completed" from code values  
ServiceRequest.doNotPerform cardinallity to 0..0  (Note: doing so removed doNotPerformed from differential table, but it remains in snapshot as 0..1  -  doNotPerform still referenced in Guidance)  

Note: ProcedureNotDone, Fixed value = "not-done" already present

**https://jira.hl7.org/browse/FHIR-39307**

Negation rationale for Assessment, Performed now references QICoreObservationCancelled
Negation rationale for Diagnostic Study, Performed now references QICoreObservationCancelled
Title for negation rationale for Immunization, Administered corrected
Negation rationale for Laboratory Test, Performed now references QICoreObservationCancelled
Title for negation rationale for Laboratory Test, Order corrected
Negation rationale for Medication, Dispensed now references QICoreMedicationDispenseDeclined
Negation rationale for Physical Exam, Performed now references QICoreObservationCancelled

New StructureDefinition profiles added:

Communication
DeviceRequest
MedicationAdministration
MedicationDispense
MedicationDispenseDeclined
ObservationCancelled
Procedure
Task
TaskRejected

### CODE REVIEW

**Comment by p9g** [comment]
Thanks for the links to and from HL7 Jira.

File: input/cql/ProfilePatterns.cql [hunk 1]
```diff
@@ -268,9 +268,9 @@ define Observation:
   [Observation] O
     where O.status in { 'registered', 'preliminary', 'final', 'amended', 'corrected' }
 
-// QICoreObservationNotDone
-// @profile: http://hl7.org/fhir/us/qicore/StructureDefinition/qicore-observationnotdone
-define ObservationNotDone:
+// QICoreObservationCancelled
+// @profile: http://hl7.org/fhir/us/qicore/StructureDefinition/qicore-observationcancelled
+define ObservationCncelled:
```
Comment (p9g) [inline input/cql/ProfilePatterns.cql:273]:
```suggestion
define ObservationCancelled:
```

**Comment by p9g** [comment]
Regarding the new profiles, https://jira.hl7.org/browse/FHIR-38788 says "StructureDefinition is required for each (in lieu of a spreadsheet resource".  Looks like there are still quite a few profiles that use spreadsheet xml, and for these there are no tabs for xml, json, and ttl views of the profile.  (For example, [QICore Encounter](http://hl7.org/fhir/us/qicore/2022Sep/StructureDefinition-qicore-encounter.html) has tabs, [QICore Patient](http://hl7.org/fhir/us/qicore/2022Sep/StructureDefinition-qicore-patient.html) does not) Are these tabs only generated when the spreadsheet is not used?  Is there a way to generate those tabs for the resources whose profiles are still using the spreadsheet approach?

File: input/cql/ProfilePatterns.cql (hunk 1 above)
Comment (dand9959) [reply input/cql/ProfilePatterns.cql]: Fixed typo.  Thanks for finding that.   Commit: b43e6b6

**Comment by dand9959** [comment]
That's a good question. I don't have an answer, though, as I'm not all that familiar yet with the tooling behavior.  I do know that moving all the profiles to StructureDefinitions is the goal - but making fixes to the IG is currently the priority.

**Comment by abdullah-git1** [comment]
Not sure why the published versions are different from the build but [QI-Core Patient](http://build.fhir.org/ig/HL7/fhir-qi-core/StructureDefinition-qicore-patient.html) does contain those tabs that are missing in the published version. As Dan mentioned, the goal is to apply all of the changes to the profiles first and then change the profile spreadsheets to StructureDefinition

**Pull request merged**